rclcpp/test/rclcpp/generic/memory_management.hpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__GENERIC__MEMORY_MANAGEMENT_HPP_
+#define RCLCPP__GENERIC__MEMORY_MANAGEMENT_HPP_
+
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rcutils/error_handling.h"
+
+namespace rclcpp_generic
+{
+class MemoryManagement
```

Comment:
This a strange class name... Also I think it should just be converted into functions. I don't see a need for a class here.

Reply:
Actually, it seems like this existing class should be enough:

https://github.com/ros2/rclcpp/blob/d9377dc740d4424d0740131e79cfdb2a31dcf87c/rclcpp/include/rclcpp/serialization.hpp#L61-L75

Though it also has issues with using pointers where it should probably use `const &` instead.

Maybe update it instead to have template methods on it rather than requiring the typesupport to be provided to it manually?

@Karsten1987 may have some input, though I think he might be out of touch through the end of the year.

Reply:
yeah, this is clearly legacy from an early rosbag2 development. It sounds right to me to leverage/modify the functionality in `serialization.hpp`.

Reply:
Thanks for pointing that out! Done.